### ISSUE TITLE
Motivating examples for the readme

### ISSUE THREAD

**Issue by goodboy** [goodboy/tractor#175]
Clearly, we need some fancier and more immediately practical examples in our readme (says every person who's glanced ever), so I figured I'd start list of what I think would be coolio.

#### A "worker pool" example copying `multiprocessing`'s [prime number calculator](https://docs.python.org/3/library/concurrent.futures.html#processpoolexecutor-example)
- it is the example for the [`ProcessPoolExecutor`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor) (ugh) so we'll need to probably mock up the [executor api](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor) and show the SC way of doing things

#### A "controlled fork blast" example
- this would be an example of creating a [fork bomb](https://en.wikipedia.org/wiki/Fork_bomb) out of nested nurseries which can be "controlled" via a SIGINT from the user once they've decided that enough CPU has been crushed; this demonstrates very succinctly the true power of SC.
- we might want to monitor system resources for this test to avoid making it impossible to send input whatsoever

#### A real-time streaming sorting algorithm
- I was thinking of implementing an async-ified version of [pythonmaxmin](https://github.com/lemire/pythonmaxmin) which is based on the paper [here](https://arxiv.org/abs/cs/0610046)
- Another thing we might be able to show off is performing this sort on multiple streams in parallel and using a `trio`-ized version of an [Rx `merge`](http://introtorx.com/Content/v1.0.10621.0/12_CombiningSequences.html#Merge) to serialize results.

Lurker opinions always welcome.

**Comment by parity3** [comment]
Wow, some of these, particularly the stream sorting may do well as their own separate repos/libraries!

Another possible example is to implement one of the function-as-a-service protocols with some tuning goodies.

**Comment by goodboy** [comment]
> Another possible example is to implement one of the function-as-a-service protocols with some tuning goodies.

@parity3 sweet idea. I'd welcome a PR for something that too 😉 

> function-as-a-service 

Also check #182, I think the caps based RPC is going to be here to stay the more I've thought about it.
I actually have been dreaming up auto-generating capn proto schema from python's type annotations.

**Comment by parity3** [comment]
Another piece of code that may be nice to see is for a tractor controller app to run as a daemon that some other sync or polyglot app could call into, and be used as a glorified task queue implementation. You could make it compatible with celery or dramatiq or whatever.

**Comment by goodboy** [comment]
> You could make it compatible with celery

I was gonna say, IMO if you pair `tractor` with `nanomessage` or `zeromq` you can pretty much implement `celery` somewhat trivially, but it'll be `rabbitmq` free / `tractor` broker kinda thing.

That might be good further down the line to demonstrate how with the right primitives from the outset these specialized message pattern designs like `celery` (which AMQP **is** just a special case of something you can build with [scalability protocols](http://wiki.zeromq.org/whitepapers:brokerless) and/or an actor model) are super easy to create without a giant slew of components / configuration.

This was kind of always the intention from the outset: make the *`zeromq` of micro-service systems* showing how most message oriented designs are just special cases if you start with the right primitives. This will be extended by theory around security principles from capability based sec and how you can build ultra-resilient supervisor trees with this approach.

**Comment by goodboy** [comment]
Hmm probably worth linking celery/ceps#27

**Comment by goodboy** [comment]
@parity3 yeah maybe a "micro-service design" tutorial is in order as well.

I think we'll end up picking the simplest / hippest of these examples and toss the rest in a "parallelism tutorial" in the docs.
